AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreObserveQueryTests.swift

```diff
@@ -104,11 +179,223 @@ class DataStoreObserveQueryTests: SyncEngineIntegrationTestBase {
                 XCTFail("\(error)")
             }
         } receiveValue: { querySnapshot in
+            snapshots.append(querySnapshot)
             if querySnapshot.isSynced {
                 snapshotWithIsSynced.fulfill()
             }
         }
+        savePostAndWaitForSync(Post(title: "title", content: "content", createdAt: .now()))
         wait(for: [snapshotWithIsSynced], timeout: 100)
+        XCTAssertTrue(snapshots.count >= 2)
+        XCTAssertFalse(snapshots[0].isSynced)
+        XCTAssertTrue(snapshots.last!.isSynced)
+        XCTAssertTrue(snapshots.last!.items.count > numberOfPosts)
+        sink.cancel()
+    }
+
+    /// ObserveQuery is set up with a query predicate.
+    /// Sync is completed and actions are applied after sync to observe expected snapshots.
+    ///
+    /// - Given: DataStore has already synced the models to local store. ObserveQuery is started with a predicate
+    /// - When/Then:
+    ///     - Add at least one model that matches the given predicate. Model is received on the snapshot.
+    ///     - Add at least one model that does NOT match the given predicate. Model is not received on the snapshot
+    ///     - Update a model that matches the given predicate so that it no longer matches the predicate.
+    ///         Model is removed from the snapshot.
+    ///     - Update a model that does NOT match the predicate so that it now matches the predicate.
+    ///         Model is added to the snapshot
+    ///     - Delete a model that matches the predicate. Model is removed from the snapshot
+    ///     - Delete a model that does NOT match the predicate. No snapshot is emitted
+    func testPredicateWithCreateUpdateDelete() throws {
+        let started = expectation(description: "Amplify started")
+        try startAmplify {
+            started.fulfill()
+        }
+        wait(for: [started], timeout: 2)
+        Amplify.Logging.logLevel = .info
+
+        let testId = UUID().uuidString
```

Comment:
nit rename to `testContent`

Reply:
the intent here was to use a predicate on the `testId` for this particular test run, so data is isolated to local models of the test. the fact that it's stored in content is not really relevant and would prefer to keep this as `testId` or `testIdentifier`, let me know if you have a preference or still think it should be `testContent`